### PR TITLE
69 implement simple api endpoint with service logic for sandboxes

### PR DESCRIPTION
**Pull request by MaximPyatigor**

### CODE REVIEW

File: SAPex/Controllers/SandboxesController.cs [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Mvc;
+using PagedList;
+using SAPex.Mappers;
+using SAPex.Models;
+using SAPex.Models.Validators;
+
+namespace SAPex.Controllers
+{
+    [Route("api/sandbox")]
```
Comment (dnavitski) [inline SAPex/Controllers/SandboxesController.cs]: The name "Sandboxes" of the controller is fine. But in the route, we have "sandbox" still. Please update the route name.

File: SAPex/Models/Validators/SandboxValidator.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace SAPex.Models.Validators
+{
+    public class SandboxValidator : AbstractValidator<SandboxViewModel>
+    {
+        public SandboxValidator()
+        {
+            // RuleFor(sandbox => sandbox.Id).NotNull();
+            RuleFor(sandbox => sandbox.Name).NotEmpty().WithMessage("Sandbox name can t be empty");
+            RuleFor(sandbox => sandbox.Description).NotEmpty();
```
Comment (dnavitski) [inline SAPex/Models/Validators/SandboxValidator.cs:11]: All these rules should be used in different cases. They can not be used for one model. Please discuss with mentors.

File: SAPex/Services/InitService.cs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using DbMigrations.Data;
+using DbMigrations.EntityModels;
+using Microsoft.EntityFrameworkCore;
+
+namespace SAPex.Services
+{
+    public class InitService
```
Comment (dnavitski) [inline SAPex/Services/InitService.cs]: Please create the "Helpers" folder in the BLL project and move the "Initial" service into this folder. Please rename class like "TestDataHelper" or "InitialDataHelper".

File: SAPex/Services/InitService.cs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        public static void Initial(string connectionString)
+        {
+            var optionsBuilder = new DbContextOptionsBuilder<AppDbContext>();
+            var options = optionsBuilder.UseSqlServer(connectionString).Options;
+
+            using (AppDbContext db = new AppDbContext(options))
+            {
+                if (!db.Sandboxes.Any())
+                {
+                    Guid id1 = new Guid("653fa839-37e8-41df-9fba-440a51466788");
```
Comment (dnavitski) [inline SAPex/Services/InitService.cs]: Please use id1 = Guid.NewGuid()

**Review by dnavitski — CHANGES_REQUESTED** [review]

File: SAPex/Controllers/SandboxesController.cs (hunk 1 above)
Comment (MaximPyatigor) [reply SAPex/Controllers/SandboxesController.cs]: done.

File: SAPex/Models/Validators/SandboxValidator.cs (hunk 2 above)
Comment (MaximPyatigor) [reply SAPex/Models/Validators/SandboxValidator.cs:11]: in progress

File: SAPex/Services/InitService.cs (hunk 3 above)
Comment (MaximPyatigor) [reply SAPex/Services/InitService.cs]: done

File: SAPex/Services/InitService.cs (hunk 4 above)
Comment (MaximPyatigor) [reply SAPex/Services/InitService.cs]: done

**Pull request closed without merging**